Core/Src/speed_calc.c

```diff
@@ -54,64 +55,47 @@ void speed_calc(void *argument) {
-			//zera a velocidade de todas as rodas
-			for(speed_pin_e i = FRONT_RIGHT; i <= REAR_LEFT; i++)
-				g_wheel_speed[i] = 0;
+			reset_speed_all();  //zera a velocidade de todas as rodas
 			break;
 
 		//caso a tarefa tenha sido chamada pela queue
 		case osOK:
 			//verifica se alguma roda está a muito tempo sem receber interrupção, caso sim a velocidade dessa roda é zerada
-			for(speed_pin_e i = FRONT_RIGHT; i <= REAR_LEFT; i++)
-				if((message.tim_count - last_messages[i].tim_count) > min_count)
-					g_wheel_speed[i] = 0;
-
+			reset_speed_single(message, last_messages, min_count);
 
-			speed_pin_e wheel_pin = get_speed_pin(message.pin);						//obtem de qual pino veio a interrupção
-			d_tim_count = message.tim_count - last_messages[wheel_pin].tim_count; 	//diferenca entre timestamp da mensagem atual e da anterior
+			//diferenca entre timestamp da mensagem atual e da anterior
+			d_tim_count = message.tim_count - last_messages[message.pin].tim_count;
 
-			if(d_tim_count < max_count) { //caso d_tim_count calcule uma velocidade maior do que é possível o valor será descartado
-				Error_Handler();
+			//caso d_tim_count calcule uma velocidade maior do que é possível o valor será descartado
+			if(d_tim_count < max_count)
 				continue;
-			}
 
+			d_tim_countx=d_tim_count;
 			speed = (10*3.6*2*M_PI * WHEEL_RADIUS / SPEED_SENSOR_TEETH_QUAN) * ((float)tim_freq/((float)d_tim_count*tim_presc)); //calcula a velocidade
-			g_wheel_speed[wheel_pin] = speed; 		//seta velocidade especifica da roda recebida
-			last_messages[wheel_pin] = message; 	//guarda mensagem até a próxima interacão
+			g_wheel_speed[message.pin] = speed;     //seta velocidade especifica da roda recebida
+			last_messages[message.pin] = message;   //guarda mensagem até a próxima interacão
 
-			uint16_t datalog_id = 	wheel_pin == FRONT_RIGHT ? 	ID_SPEED_FR :
-									wheel_pin == FRONT_LEFT  ? 	ID_SPEED_FL :
-									wheel_pin == REAR_RIGHT  ? 	ID_SPEED_RR :
-																ID_SPEED_RL;
+			uint16_t datalog_id =   message.pin == FRONT_RIGHT ?    ID_SPEED_FR :
+			                        message.pin == FRONT_LEFT  ?    ID_SPEED_FL :
+			                        message.pin == REAR_RIGHT  ?    ID_SPEED_RR :
+			                                                        ID_SPEED_RL;
 			log_data(datalog_id, (uint16_t) speed);
 		break;
 		}
 	}
 }
 
-
-
-speed_pin_e get_speed_pin(uint16_t pin){
-	switch(pin){
-	case S_VEL1_Pin:
-		return FRONT_RIGHT;
-		break;
-	case S_VEL2_Pin:
-		return FRONT_LEFT;
-		break;
-	case S_VEL3_Pin:
-		return REAR_RIGHT;
-		break;
-	default:
-		return REAR_LEFT;
-		break;
-	}
+void reset_speed_all(){
+	for(speed_pin_e i = FIRST_WHEEL; i <= LAST_WHEEL; i++)
+		g_wheel_speed[i] = 0;
 }
 
-
-
-
+void reset_speed_single(speed_message_t message, speed_message_t last_messages[4], uint32_t min_count){
+	for(speed_pin_e i = FIRST_WHEEL; i <= LAST_WHEEL; i++)
+		if((message.tim_count - last_messages[i].tim_count) > min_count)
+			g_wheel_speed[i] = 0;
+}
```

Comment:
Muda aqui agr q é ponteiro as coisas
```suggestion
void reset_speed_single(speed_message_t* message, speed_message_t last_messages*, uint32_t min_count){
	for(speed_pin_e i = FIRST_WHEEL; i <= LAST_WHEEL; i++)
		if((message->tim_count - last_messages[i].tim_count) > min_count)
			g_wheel_speed[i] = 0;
}
```